### DISCUSSION TITLE
My PDFs are getting errors when being opened in Adobe Acrobat. Am I doing something wrong when adding images?

### DISCUSSION THREAD

**Discussion by Decimator1** [QuestPDF/QuestPDF#983] [Q&A]
I have been creating PDF files that contain several hundred PNG files.

When I create the document, it opens without issue in any web browser, but when I open it in Adobe Acrobat 2020, I get the following error and Adobe does not draw the images.

![insufficientdata](https://github.com/user-attachments/assets/056a5713-e7fd-4879-9ad6-5d6879243508)

However, if I use the UseOriginalImage() option with my image adding code, Adobe can handle the images without issue.

Is this a known issue with Adobe, or am I doing something wrong when adding the images?

This is the code snippet that adds images to the PDF document.

```
foreach (var note in _printCaseDetails.caseNotes)
{
    Uri frontUri = new Uri(note.FrontImagePath); 
    var frontImagePath = Path.Combine(@"C:\\Images\\", Path.GetFileName(frontUri.LocalPath));

    if (File.Exists(frontImagePath) == false) 
    {
        caseNotesTable.Cell().Element(NoteBlock).Image(noImage);
    }
    else
    {
        //var frontImage = QuestPDF.Infrastructure.Image.FromFile(frontImagePath);
        caseNotesTable.Cell().Element(NoteBlock).Image(frontImagePath);
    }

    Uri backUri = new Uri(note.BackImagePath);
    var backImagePath = Path.Combine(@"C:\\Images\\", Path.GetFileName(backUri.LocalPath));

    if (File.Exists(backImagePath) == false)
    {
        caseNotesTable.Cell().Element(NoteBlock).Image(noImage);
    }
    else
    {
        caseNotesTable.Cell().Element(NoteBlock).Image(backImagePath);
    }
}
```

And this is the NoteBlock element I use for creating the cells:
```
static IContainer NoteBlock(IContainer container)
{
    return container
        .Border(1)
        .Background(Colors.White)
        .ShowOnce()
        .PaddingLeft(5)
        .MinWidth(50)
        .MinHeight(50)
        .AlignLeft()
        .AlignMiddle();
}
```

I have made no changes to the DocumentMetadata or DocumentSettings

**Comment by BeneHenke** [comment]
It seems like an issue with adobe acrobat. The way you add the images seems fine. Can you try a newer version of adobe?

**Comment by MarcinZiabek** [comment]
Could you also try with a different image? Maybe yours is broken somehow and becomes unreadable after resizing/compressing. Many web browsers may still accept and display such an image.

**Comment by Decimator1** [comment]
The same thing occurred for the vast majority of the images I was using when creating documents. A few of them did load successfully in other documents, which leads me towards the broken image theory. Unfortunately, I cannot use different images since they are tied to specific records in the PDF. 

Does QuestPDF have any methods to reduce the PDF size after creation, like how Adobe has their "Save as Optimized PDF" option? Saving the PDF with that option drastically reduces the PDF file size while leaving the images intact.

**Comment by DawidRobertNowak** [comment]
Hello,

We are closing this thread due to inactivity. We apologize for the delayed response.

Since the original message, we have applied several improvements to the image handling logic, and we have not received similar bug reports. It is likely that this has already been resolved in a more recent version.

Should you encounter any further problems, please don't hesitate to open a new issue.
Thank you!
